.azure-pipelines/advanced.yml

```diff
@@ -11,8 +11,15 @@ schedules:
       - master
     always: true
 
-jobs:
-  # Any addition here should be reflected in the advanced-test and release pipelines.
-  # It is advised to declare all jobs here as templates to improve maintainability.
-  - template: templates/tests-suite.yml
-  - template: templates/installer-tests.yml
+stages:
+  - stage: Test_Package
```

Comment:
I think `TestAndPackage` is the best option here.

On the alternatives.

For 1) it is not helping for the main purpose, since the name of the stage is used in the header displayed in Azure Pipeline UI of each job running inside it.

For 2), I think it adds unnecessary complexity to the pipeline(s), since every job from test and package can run in parallel. Defining stages make them running sequentially by default, and you need to relax this behavior explicitly in the pipeline(s). I prefer keeping it simple and put every parallelizable job in the same stage.

Indeed, refactoring `TestAndPackage` into one stage remove the need to explicitly synchronize the serie of jobs for that stage in each pipeline where it is needed. I moved the changelog into a dedicated stage: in theory the job could be run in parallel with the jobs from `TestAndPackage`, but the build is fast, and it is not a big deal to loose 1 min in the release pipeline.